### ISSUE TITLE
No codeLens results when using stable toolchain

### ISSUE THREAD

**Issue by martskins** [rust-analyzer/rust-analyzer#2665]
I'm running into some weird issue with CodeLens and different (or not so) versions of rust. I created a dummy project to test the codeLens feature on vim and it works with 1.39 and 1.40, but it doesn't seem to do so if I use `stable` instead of 1.40, which should be the same afaik.

The project was created with `cargo new whatever` and the code is only a few lines in `main.rs`:

```
struct A {}

impl A {
    fn new() -> Self {
        A {}
    }
}

fn main() {
    println!("Hello, world!");
}
```
If I run `rustup override set 1.39.0` or `rustup override set 1.40.0` I get results from CodeLens for both the struct and `fn main`, as expected. But if I run `rustup override set stable` (which sets rustc to the same exact version as `rustup override set 1.40.0`) I get no results back from it, no error either, just `null`.

**Comment by bjorn3** [comment]
What does `cargo +stable metadata` show? (if json, please just say so instead of copying all json. if an error, please do copy it all)

**Comment by martskins** [comment]
This warning: `warning: please specify '--format-version' flag explicitly to avoid compatibility problems` and JSON

**Comment by matklad** [comment]
Wild guess: could it be that `rust-src` component is installed for one, but not for the other, toolchain?

**Comment by martskins** [comment]
It shows as up to date for both 1.40.0 and stable (which makes sense), also I believe if that was the issue I would get no completion neither, but I do get completion just fine.

**Comment by lnicola** [comment]
@martskins does this still happen?

![image](https://user-images.githubusercontent.com/308347/102769768-ca640600-438b-11eb-87fb-efeb7e4ff165.png)

**Comment by martskins** [comment]
Oh sorry, forgot to close this. No, it doesn't happen anymore, not sure what resolved it but it's definitely working for me now.
